src/main/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesService.java

```diff
@@ -1,64 +1,40 @@
 package uk.gov.digital.ho.pttg.application.namematching;
 
-import com.google.common.collect.Lists;
-import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Service;
 import uk.gov.digital.ho.pttg.application.namematching.candidates.MultipleLastNames;
 import uk.gov.digital.ho.pttg.application.namematching.candidates.NameCombinations;
+import uk.gov.digital.ho.pttg.application.namematching.candidates.SpecialCharacters;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 import static uk.gov.digital.ho.pttg.application.namematching.NameMatchingFunctions.deduplicate;
 
 @Service
 public class NameMatchingCandidatesService {
-    private static final String NAME_SPLITTERS = "-'";
-    private static final String NAME_SPLITTER_REGEX = "[" + NAME_SPLITTERS + "]";
 
     private NameCombinations nameCombinations;
     private MultipleLastNames multipleLastNames;
+    private SpecialCharacters specialCharacters;
 
-    public NameMatchingCandidatesService(NameCombinations nameCombinations, MultipleLastNames multipleLastNames) {
+    public NameMatchingCandidatesService(NameCombinations nameCombinations, MultipleLastNames multipleLastNames, SpecialCharacters specialCharacters) {
```

Comment:
Can this be changed so the constructor takes a List of NameMatchingGenerators as a single parameter? Then we could add new generators without having to wire them into the generateCandidateNames method. I don't know if we'd lose control over the ordering though. If so then maybe an improvement when we introduce scoring?

Reply:
Ordering is the problem.  Let's discuss.

Reply:
We'll leave it like this until scoring is implemented